GPDB DOCS - DO command - anonymous blocks

mkiyama opened the pull request:
Update GPDB 5.0 docs for DO command and anonymous block examples.
Also updated navigation and feature summary files for DO command and other GPDB 5.0 updates.

hlinnaka commented on gpdb-doc/dita/ref_guide/extensions/pl_sql.xml:
```diff
@@ -140,8 +143,8 @@ $$ LANGUAGE plpgsql;</codeblock>
       <body>
-        <p>Parameters passed to functions are named with the identifiers such asa
-            <codeph>$1</codeph>, <codeph>$2</codeph>. Optionally, aliases can be declared for
+        <p>Parameters passed to functions are named with the identifiers such as
```
should we drop the "the" here? As in, "... are named with identifiers such as ..."

mkiyama replied:
thanks for catching the typo.

hlinnaka commented on gpdb-doc/dita/ref_guide/feature_summary.xml:
```diff
@@ -565,14 +575,21 @@ NEXT 10 ROWS ONLY; </codeblock><p>Greenplum
               <entry colname="col3"><b>Unsupported Clauses /
-                    Options:</b><p><codeph>RETURNING</codeph></p></entry>
+                    Options:</b><p><codeph>RETURNING</codeph></p><p><b>Limitations:</b></p><p><codeph>UPDATE/DELETE
+                    WHERE CURRENT OF...</codeph> cursor statements must be executed by Greenplum
```
I don't understand what this limitation means. Aren't all statements executed by Greenplum Database?

hlinnaka replied:
I just saw that there's a separate PR for this (https://github.com/greenplum-db/gpdb/pull/1980). We can discuss this there.

mkiyama replied:
I'll work with Chuck clarify the text.

hlinnaka commented on gpdb-doc/dita/ref_guide/sql_commands/DO.xml:
```diff
@@ -0,0 +1,109 @@
+  RAISE NOTICE 'user % is %a superuser', t1_row.rolname, superuser ;
+END $$ LANGUAGE plpgsql ;</codeblock></p>
+    </section>
+    <note type="note">The example PL/pgSQL uses <codeph>SELECT</codeph> with the
```
Perhaps we should avoid using the SELECT INTO construct, so that we don't need to explain the difference here?

mkiyama replied:
@hlinnaka 
I think the example is "useful" and the note is a useful "heads up".
Is there another way to write the anon block?

hlinnaka replied:
It's a useful heads-up, I totally agree with that. But it's useful for PL/pgSQL development in general, not related to anonymous blocks.

You could use a subquery, like:
  rolsuper = (SELECT rolsuper FROM pg_authid, list WHERE pg_authid.rolname = name);

I find that easier to read than SELECT INTO, precisely because SELECT INTO is easy to confuse with the other SELECT INTO. I'm not sure if it behaves the same if no row is found, though.

Anyway, I don't feel strongly about it, just thought it looked a bit out-of-place.

lisakowen commented:
are anonymous blocks supported just for sql/plpgsql?  or do pl/python and pl/perl support also anonymous blocks?  if they do, might want to note that on their respective "Greenplum PL/* Language Extension" page.

mkiyama commented:
@lisakowen 
I had a question out to Xiong Gang
GPDB only supports PL/pgSQL (plpgsql), PL/Python (plpythonu), and PL/Perl (plperl and plperlu)

I'll update the docs.

lisakowen commented:
maybe this will answer my above question - does "CREATE LANGUAGE" now support the [INLINE inline_handler] for anonymous blocks?  i.e. is anonymous block support something that can be turned on/off for other PLs when you create the language, or are they always supported?

mkiyama commented:
@hlinnaka would you be able to answer @lisakowen  question?

Also, to extend the question, has GPDB CREATE LANGUAGE been updated to postgres 9.0?
Would this mean that the CREATE LANGUAGE docs would also need to be updated to 9.0
to match the Postgres docs?

If so, are there other changes, like catalog changes?

mkiyama commented:
Updated docs based on review. 
- Remove the cursor limiation. That will be addressed in another commit.
- Added anonymous block information to the PL/Python section.

dyozie commented on gpdb-doc/dita/ref_guide/sql_commands/DO.xml:
```diff
@@ -0,0 +1,112 @@
+        anonymous function in a procedural language.</p>
+      <p>The code block is treated as though it were the body of a function with no parameters,
+        returning void. It is parsed and executed a single time.</p>
+      <p>The optional <codeph>LANGUAGE</codeph> clause can be either before or after the code
```
Change "be" to "appear"

dyozie commented on gpdb-doc/dita/ref_guide/sql_commands/DO.xml:
```diff
@@ -0,0 +1,112 @@
+          <pd>The procedural language code to be executed. This must be specified as a string
+            literal, just as with the <codeph>CREATE FUNCTION</codeph> command. Use of a
+            dollar-quoted literal is recommended. Optional keywords have no effect. These procedural
+            language are supported: PL/pgSQL (<codeph>plpgsql</codeph>), PL/Python
```
"language" -> "languages"

dyozie commented on gpdb-doc/dita/ref_guide/sql_commands/DO.xml:
```diff
@@ -0,0 +1,112 @@
+            dollar-quoted literal is recommended. Optional keywords have no effect. These procedural
+            language are supported: PL/pgSQL (<codeph>plpgsql</codeph>), PL/Python
+              (<codeph>plpythonu</codeph>), and PL/Perl (<codeph>plperl</codeph> and
+              <codeph>plperlu</codeph>)</pd>
```
Missing punctuation here.

mkiyama commented:
Fixed typos.

mkiyama commented:
created a commit to fix typos.

mkiyama commented:
updated feature repo w/ gpdb repo commits.

mkiyama commented:
Updated Ref. Guide topics for DO command. (CREATE LANGUAGE command and pg_language catalog table)
